src/oss-defog/DefoggerTool.cs

```diff
@@ -213,6 +421,16 @@ public void AnalyzeFile(string filename)
             var fileContents = File.ReadAllText(filename);
 #pragma warning restore SEC0116 // Path Tampering Unvalidated File Path
 
+            AnalyzeFile(filename, fileContents);
+        }
+
+        public bool HasNonTextContent(string content)
+        {
+            return content.Any(ch => char.IsControl(ch) && ch != '\t' && ch != '\r' && ch != '\n');
```

Comment:
Might be unnecessary, but [Char.IsWhiteSpace](https://docs.microsoft.com/en-us/dotnet/api/system.char.iswhitespace?view=net-5.0) handles unicode whitespace too.

Reply:
I think that should work.  I was being more restrictive because whitespace might mean like vertical tab which doesn't make sense in the context of code.  I'll just use the built in instead though.